test/libsolidity/util/compiler/InternalCompiler.cpp

```diff
@@ -0,0 +1,180 @@
+using namespace solidity::frontend::test;
+
+CompilerOutput InternalCompiler::compile(CompilerInput const& _input)
+{
+    configure(_input);
+
+    auto [contracts, errors] = internalCompile(_input);
+    bool success = m_stack.compilationSuccessful();
+
+    return CompilerOutput{
+        contracts,
+        success,
+        errors,
+        formatErrorInformation()
+    };
+}
+
+void InternalCompiler::configure(CompilerInput const& _input)
+{
+    m_stack.reset();
+
+    m_stack.setSources(_input.sourceCode);
+    m_stack.setLibraries(_input.libraryAddresses);
+
+    if (_input.evmVersion.has_value())
+        m_stack.setEVMVersion(_input.evmVersion.value());
+    if (_input.optimise.has_value())
+        m_stack.setOptimiserSettings(_input.optimise.value());
+    if (_input.optimiserSettings.has_value())
+        m_stack.setOptimiserSettings(_input.optimiserSettings.value());
+    if (_input.revertStrings.has_value())
+        m_stack.setRevertStringBehaviour(_input.revertStrings.value());
+    if (_input.metadataFormat.has_value())
+    {
+        CompilerStack::MetadataFormat metadata = CompilerStack::MetadataFormat::NoMetadata;
+        switch (_input.metadataFormat.value())
+        {
+            case MetadataFormat::WithReleaseVersionTag:
+                metadata = CompilerStack::MetadataFormat::WithReleaseVersionTag;
+                break;
+            case MetadataFormat::WithPrereleaseVersionTag:
+                metadata = CompilerStack::MetadataFormat::WithPrereleaseVersionTag;
+                break;
+            default:
+                break;
+        }
+        m_stack.setMetadataFormat(metadata);
+    }
+    if (_input.metadataHash.has_value())
+    {
+        CompilerStack::MetadataHash hash = CompilerStack::MetadataHash::None;
+        switch (_input.metadataHash.value())
+        {
+            case MetadataHash::IPFS:
+                hash = CompilerStack::MetadataHash::IPFS;
+                break;
+            case MetadataHash::Bzzr1:
+                hash = CompilerStack::MetadataHash::Bzzr1;
+                break;
+            default:
+                break;
+        }
+        m_stack.setMetadataHash(hash);
+    }
+    if (_input.viaIR.has_value())
+        m_stack.setViaIR(_input.viaIR.value());
+    m_stack.setEOFVersion(_input.eofVersion);
+}
+
+std::pair<MappedContracts, Errors> InternalCompiler::internalCompile(
+    CompilerInput const& _input
+)
+{
+    m_stack.compile();
+
+    MappedContracts mappedContracts{};
+    for (auto sourceName: m_stack.sourceNames())
+    {
+        std::vector<CompiledContract> compiled;
+        for (auto contract: m_stack.contractDefinitions(sourceName))
+        {
+            std::string contractName = contract->fullyQualifiedName();
+            LinkerObject object = m_stack.object(contractName);
+            LinkerObject runtimeObject = m_stack.runtimeObject(contractName);
+            bool hasUnlinkedReferences = !object.linkReferences.empty();
```

Comment:
this is unused

Reply:
Could you elaborate on this? The variable is used further down when constructing `CompiledContract`. And the member is then checked in `SolidityExecutionFramework::multiSourceCompileContract()`.

Reply:
my bad :sweat_smile: